java-frontend/src/main/java/org/sonar/java/resolve/SecondPass.java

```diff
@@ -113,6 +114,22 @@ private void complete(JavaSymbol.TypeJavaSymbol symbol) {
     if ((symbol.flags() & Flags.INTERFACE) == 0) {
       symbol.members.enter(new JavaSymbol.VariableJavaSymbol(Flags.FINAL, "super", type.supertype, symbol));
     }
+
+    // Register default constructor
+    if (tree.is(Tree.Kind.CLASS) && symbol.lookupSymbols("<init>").isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "<init>" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1192)